### PR TITLE
Lasb 1014 maat court data api add new entities and repository classes for create assessment history

### PR DESCRIPTION
**Pull request by venkat980**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/LASB-1014)

Add entity and repository classes for create_assessment_history

## Checklist

Before you ask people to review this PR:

- [x] Tests should be passing: `./gradlew test`
- [x] Github should not be reporting conflicts; you should have recently run `git rebase main`.
- [x] Avoid mixing whitespace changes with code changes in the same commit. These make diffs harder to read and conflicts more likely.
- [x] You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- [x] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: maat-court-data-api/src/main/java/gov/uk/courtdata/entity/ChildWeightHistoryEntity.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import lombok.*;
+import org.hibernate.annotations.CreationTimestamp;
+
+import javax.persistence.*;
+import java.time.LocalDateTime;
+
+@Builder
+@AllArgsConstructor
+@NoArgsConstructor
+@Getter
```
Comment (atharmajeed) [inline maat-court-data-api/src/main/java/gov/uk/courtdata/entity/ChildWeightHistoryEntity.java:12]: this '@Data' annotation covers most annotation like getter and setter etc.

File: maat-court-data-api/src/main/java/gov/uk/courtdata/entity/FinancialAssessmentDetailsHistoryEntity.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import lombok.*;
+
+import javax.persistence.*;
+import java.math.BigDecimal;
+import java.time.LocalDate;
+
+@Builder
+@AllArgsConstructor
+@NoArgsConstructor
+@Getter
```
Comment (atharmajeed) [inline maat-court-data-api/src/main/java/gov/uk/courtdata/entity/FinancialAssessmentDetailsHistoryEntity.java:13]: same here '@Data annotation'

File: maat-court-data-api/src/main/java/gov/uk/courtdata/repository/ChildWeightHistoryRepository.java [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package gov.uk.courtdata.repository;
+
+import gov.uk.courtdata.entity.ChildWeightHistoryEntity;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+public interface ChildWeightHistoryRepository extends JpaRepository<ChildWeightHistoryEntity, Integer> {
+//    List<ChildWeightHistoryEntity> findAllByFinancialAssessmentId(Integer financialAssessmentId);
```
Comment (atharmajeed) [inline maat-court-data-api/src/main/java/gov/uk/courtdata/repository/ChildWeightHistoryRepository.java:7]: we can remove this unused code.

File: maat-court-data-api/src/main/java/gov/uk/courtdata/repository/FinancialAssessmentDetailsHistoryRepository.java [hunk 4]
```diff
@@ -0,0 +1,8 @@
+package gov.uk.courtdata.repository;
+
+import gov.uk.courtdata.entity.FinancialAssessmentDetailsHistoryEntity;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+public interface FinancialAssessmentDetailsHistoryRepository extends JpaRepository<FinancialAssessmentDetailsHistoryEntity, Integer> {
+//    List<FinancialAssessmentDetailsHistoryEntity> findAllByFinancialAssessmentId(Integer financialAssessmentId);
```
Comment (atharmajeed) [inline maat-court-data-api/src/main/java/gov/uk/courtdata/repository/FinancialAssessmentDetailsHistoryRepository.java:7]: same, unused code.

**Review by atharmajeed — COMMENTED** [review]
Added a few minor comments.

File: maat-court-data-api/src/main/java/gov/uk/courtdata/entity/ChildWeightHistoryEntity.java (hunk 1 above)
Comment (venkat980) [reply maat-court-data-api/src/main/java/gov/uk/courtdata/entity/ChildWeightHistoryEntity.java]: Done

**Review by atharmajeed — APPROVED** [review]

**Pull request merged**